src/index.js

```diff
@@ -143,6 +143,25 @@ Promise.prototype.then = function(onFulfilled, onRejected) {
   return prom;
 };
 
+Promise.prototype['finally'] = function(callback) {
+  if (typeof callback !== 'function') {
```

Comment:
Sorry, I made a mistake on this. This isn't enough to find all callable types. Here is the correct implementations https://github.com/npetruzzelli/es-abstract-is-callable/blob/master/lib/es2017-v8/IsCallable.js .

I don't think writing that much code is worth it (for the goal of this library being a small library). I think removing this `if` statement is a better idea. It will obviously throw an error if someone passes a non callable, but I prefer to support different types of callables.

Sorry to ask this to be changed back, I didn't fully understand the tradeoff.